server/schedule/placement/rule_manager.go

```diff
@@ -34,20 +34,22 @@ import (
 // RuleManager is responsible for the lifecycle of all placement Rules.
 // It is thread safe.
 type RuleManager struct {
-	store *core.Storage
+	storage *core.Storage
 	sync.RWMutex
 	initialized bool
 	ruleConfig  *ruleConfig
 	ruleList    ruleList
 
 	// used for rule validation
-	keyType string
+	keyType   string
+	getStores func() []*core.StoreInfo
 }
 
 // NewRuleManager creates a RuleManager instance.
-func NewRuleManager(store *core.Storage) *RuleManager {
+func NewRuleManager(storage *core.Storage, getStores func() []*core.StoreInfo) *RuleManager {
```

Comment:
I think it's better to use `StoreSetInformer` interface here.

Reply:
Addressed.